acoular/spectra.py

```diff
@@ -384,17 +399,28 @@ def calc_csm(self):
         numfreq = int(self.block_size / 2 + 1)
         csm_shape = (numfreq, t.numchannels, t.numchannels)
         csm_upper = zeros(csm_shape, dtype=self.precision)
-        # print "num blocks", self.num_blocks
+        # print( "num blocks", self.num_blocks)
         # for backward compatibility
         if self.calib and self.calib.num_mics > 0:
             if self.calib.num_mics == t.numchannels:
                 wind = wind * self.calib.data[newaxis, :]
             else:
                 raise ValueError('Calibration data not compatible: %i, %i' % (self.calib.num_mics, t.numchannels))
         # get time data blockwise
+        i = 0
         for data in self.get_source_data():
+            i+=1
             ft = fft.rfft(data * wind, None, 0).astype(self.precision)
             calcCSM(csm_upper, ft)  # only upper triangular part of matrix is calculated (for speed reasons)
+            if i == self.num_blocks and self._should_yield:
+                i = 0
+                return self._yield_csm(csm_upper, weight)
+        return self._full_csm(csm_upper, weight)
+    
+    def _yield_csm(self, csm_upper, weight):
```

Comment:
This gives a 'one time' generator which is not compatible to anything located downstream, therefore this change should be reverted.

Reply:
Reverted